test/test_other.py

```diff
@@ -7297,17 +7297,12 @@ def test_disable_inlining(self):
 ''')
 
     # Without the 'INLINING_LIMIT', -O2 inlines foo()
-    cmd = [EMCC, '-c', 'test.c', '-O2', '-o', 'test.o', '-sINLINING_LIMIT', '-flto']
+    cmd = [EMCC, 'test.c', '-O2', '-o', 'test.wasm', '-c', '-sINLINING_LIMIT', '-g']
```

Comment:
But `-O2` runs `wasm-opt`.. why doesn't wasm-opt inline the function?

Reply:
It does, that's the point of the test. Without -sINLINING_LIMIT, the function gets inlined, and then DCEd by the linker. with -sINLINING_LIMIT it does not get inlined and appears in the linked binary.

Reply:
But `-sINLINING_LIMIT` doesn't effect wasm-opt.. so how is it ever still in the binary at `-O2`?

Reply:
(`-sINLINING_LIMIT` only effect compilation not linking.. AFAICT)

Reply:
Oh, I just realized that I removed too much from this PR after I cleaned it up, and there's not even any linking step in it right now because I forgot to remove. the `-c` from the emcc command line. 
So, taking a step back, we don't actually need to link anything. The test is intended to ensure that -sINLINING_LIMIT prevents inlining at compile time. So if we make foo static, we can rely on `-O2` removing it at compile time when it's inlined, and rely on it being there when it's not inlined. And that's what the current version of this code is actually doing, it was just by accident :D